Sources/SwiftGraphQLClient/Client/Selection.swift

```diff
@@ -90,6 +90,17 @@ extension GraphQLClient {
             .tryMap { result in try result.decode(selection: selection) }
             .eraseToAnyPublisher()
     }
+
+
+    /// Executes a query request with given execution parameters.
+    public func query<T, TypeLock>(
+        _ selection: Selection<T, TypeLock>,
+        as operationName: String? = nil,
+        request: URLRequest? = nil,
+        policy: Operation.Policy = .cacheFirst
+    ) async throws -> DecodedOperationResult<T> where TypeLock: GraphQLHttpOperation {
+        try await self.query(selection, as: operationName, request: request, policy: policy).first()
+    }
```

Comment:
if `policy` is `.cacheAndNetwork` then there are two values sent by publisher, right?
I dont think this API handles that.

Reply:
That's a fair question. I'll need to test that out. 

Tbh I'm equally unsure if simply "wrapping" the APIs is the best solution here since some of the behaviour could differ – due things like this. Essentially the question for me is whether or not the "API" would be 'predictable' and would the consumer have the right expectations as to what result can be expected at all times.

How "correct" do we feel this approach is as-is @pokryfka / @maticzav ?

If its close enough (or better) I'd like to propose a 2 stage release on this.

- 5.x with this PR, to introduce wrappers around Combine (ensuring we have async support sooner rather than later)
- v6 introducing an async-backed implementation, with Combine wrappers implemented via a new package, allowing for opt-out where the developer doesn't require them

> Essentially inverting the implementation with potentially matching APIs

What's your feeling on this approach? Of course the library doesn't _have to be_ 'async first', but given one of goals of this library is to bring the best of Swift, I'd opt for that personally.

----

Again, I'd love to get this PR in ASAP though, if its "correct" 👍

Reply:
FYI, I'm planning on working on this today, so I'll try and pull in some demos and documentation – but I'd love your input as to the correctness, as I'm still far less familiar at this point. Thank!

Reply:
Thank you so much for taking the initiative to do this 🙇

My 2 cents: moving to Swift async sounds superb!